spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -181,7 +187,21 @@ class DeltaCatalog extends DelegatingCatalogExtension
       operation.mode,
       writer,
       operation,
-      tableByPath = isByPath).run(spark)
+      tableByPath = isByPath,
+      // We should invoke the Spark catalog plugin API to create the table, to
+      // respect third party catalogs. Note: only handle CREATE TABLE for now, we
+      // should support CTAS later.
+      // TODO: Spark `V2SessionCatalog` mistakenly treat tables with location as EXTERNAL table.
+      //       Before this bug is fixed, we should only call the catalog plugin API to create tables
+      //       if UC is enabled to replace `V2SessionCatalog`.
+      createTableFunc = if (isUnityCatalog && sourceQuery.isEmpty) {
```

Comment:
Does it need to be specific for unitycatalog?

Reply:
It has to, until Spark 3.5.3 is released.

Reply:
Got it. I’d only suggest adding a TODO comment. Thanks

Reply:
There is already a TODO for it.